### ISSUE TITLE
Replace packet UUID with monotonic packet sequence number

### ISSUE THREAD

**Issue by tgeoghegan** [abetterinternet/prio-server#17]
Eventually share processors have to evaluate a sequence of (own validation, peer validation, data share) triples, and must ensure that all three packets have the same UUIDs. Can the share processor assume that packets will appear in the same order in all three sequences of packets? If not, then I have to sort each packet sequence lexicographically by UUID in O(n log n) before I even begin processing triples. If instead we require that both share processors maintain the order of packets emitted by the ingestor, we can skip that step. Further, if we replace the packet UUID with a sequence number, then share processors can easily defend themselves against malformed peer validation files by verifying that sequence numbers increase monotonically as they process validation packets.

The pair (batch_uuid, packet sequence number) remains unique globally.

**Comment by tgeoghegan** [comment]
@yuriks -- thoughts on this? [This](https://github.com/abetterinternet/prio-server/blob/6ec21d09b46ccd5db4475ad5fa654e6ee309bf3f/facilitator/src/aggregation.rs#L247) is the code I was writing that led me to this idea.

**Comment by tgeoghegan** [comment]
Having thought about this more, I think the right thing to do here is to expect that the order of UUIDs as they are in the ingestion batch will be maintained. It increasingly seems as though the `prio-server` implementation will be used for both PHA server and facilitator, so we can reasonably provide that guarantee, and relying on it allows individual servers to avoid the O(n log n) sort I described earlier. So let's keep the UUID. Closing this.
